File: 0899-orderly-queue/0899-orderly-queue.py
class Solution:
    def orderlyQueue(self, s: str, k: int) -> str:
        if k==1:
            mins = s
            for i in range(len(s)):
                if s<mins:
                    mins=s
                s = s[1:]+s[0]
            return mins
        else:
            return "".join(sorted(s))

File: 0899-orderly-queue/0899-orderly-queue.cpp
class Solution {
public:
    string orderlyQueue(string S, int k) {
        int i,n;
        n = S.length();
        
        if(k == 1){
            string minString = S;
            for(i=0 ; i<n ; i++){
                string currentString;
                currentString += S.substr(i+1,(n-i-1));
                currentString += S.substr(0,(i+1));
                                
                minString = min(minString,currentString);
            }
            
            return minString;
        }
        else{
            sort(S.begin(),S.end());
            return S;                
        }
    }
};